packet/create.go

```diff
@@ -137,38 +137,38 @@ func CreatePacketWithPayload(pid uint16, cc uint8, pay []byte) Packet {
 	return pkt
 }
 func setPid(pkt *Packet, pid uint16) {
-	(*pkt)[1] = byte(pid >> 8 & 0x1f)
-	(*pkt)[2] = byte(pid & 0xff)
+	pkt[1] = byte(pid >> 8 & 0x1f)
+	pkt[2] = byte(pid & 0xff)
 }
 
 // WithHasPayloadFlag is an option function for creating a packet with a payload flag
 func WithHasPayloadFlag(pkt *Packet) {
-	(*pkt)[3] = byte((*pkt)[3] | 0x10)
+	pkt[3] |= 0x10
```

Comment:
I didn't even know this was a thing. :)

Reply:
Array pointers are indexable - this is very useful for `range` where otherwise you'd copy.